Security/TwoFactor/Provider/TwoFactorProviderRegistry.php

```diff
@@ -24,15 +27,33 @@ class TwoFactorProviderRegistry implements AuthenticationHandlerInterface
     private $providers;
 
     /**
+     * Event dispatcher.
+     *
+     * @var EventDispatcherInterface
+     */
+    protected $eventDispatcher;
+
+    /**
+     * Request parameter name used for code.
+     *
+     * @var string
+     */
+    protected $authRequestParameter;
+
+    /**
      * Initialize with an array of registered two-factor providers.
      *
-     * @param SessionFlagManager $flagManager
-     * @param array              $providers
+     * @param SessionFlagManager       $flagManager
+     * @param array                    $providers
+     * @param EventDispatcherInterface $eventDispatcher
+     * @param string                   $authRequestParameter
      */
-    public function __construct(SessionFlagManager $flagManager, $providers = array())
+    public function __construct(SessionFlagManager $flagManager, $providers = [], EventDispatcherInterface $eventDispatcher = null, $authRequestParameter = '_auth_code')
```

Comment:
If I put $eventDispatcher and $authRequestParameter before $providers and without default values, I will create a BC, no? If someone have created a subclass and call parent::__construct?